include/bluetooth/audio/audio.h

```diff
@@ -1,9 +1,1690 @@
+ */
+#define BT_CODEC_QOS_IN_UNFRAMED(_interval, _sdu, _rtn, _latency, _pd) \
+	BT_CODEC_QOS(BT_CODEC_QOS_IN, _interval, BT_CODEC_QOS_UNFRAMED, \
+		     BT_CODEC_QOS_2M, _sdu, _rtn, _latency, _pd)
+
+/** @def BT_CODEC_QOS_OUT_UNFRAMED
+ *  @brief Helper to declare Output Unframed bt_code *
+ *  @param _interval SDU interval (usec)
+ *  @param _sdu Maximum SDU Size
+ *  @param _rtn Retransmission number
+ *  @param _latency Maximum Transport Latency (msec)
+ *  @param _pd Presentation Delay (usec)
+ */
+#define BT_CODEC_QOS_OUT_UNFRAMED(_interval, _sdu, _rtn, _latency, _pd) \
+	BT_CODEC_QOS(BT_CODEC_QOS_OUT, _interval, BT_CODEC_QOS_UNFRAMED, \
+		     BT_CODEC_QOS_2M, _sdu, _rtn, _latency, _pd)
+
+/** @def BT_CODEC_QOS_INOUT_UNFRAMED
+ *  @brief Helper to declare Input/Output Unframed bt_codec_qos
+ *
+ *  @param _interval SDU interval (usec)
+ *  @param _sdu Maximum SDU Size
+ *  @param _rtn Retransmission number
+ *  @param _latency Maximum Transport Latency (msec)
+ *  @param _pd Presentation Delay (usec)
+ */
+#define BT_CODEC_QOS_INOUT_UNFRAMED(_interval, _sdu, _rtn, _latency, _pd) \
+	BT_CODEC_QOS(BT_CODEC_QOS_INOUT, _interval, BT_CODEC_QOS_UNFRAMED, \
+		     BT_CODEC_QOS_2M, _sdu, _rtn, _latency, _pd)
+
+/** @def BT_CODEC_QOS_IN_FRAMED
+ *  @brief Helper to declare Input Framed bt_codec_qos
+ *
+ *  @param _interval SDU interval (usec)
+ *  @param _sdu Maximum SDU Size
+ *  @param _rtn Retransmission number
+ *  @param _latency Maximum Transport Latency (msec)
+ *  @param _pd Presentation Delay (usec)
+ */
+#define BT_CODEC_QOS_IN_FRAMED(_interval, _sdu, _rtn, _latency, _pd) \
+	BT_CODEC_QOS(BT_CODEC_QOS_IN, _interval, BT_CODEC_QOS_FRAMED, \
+		     BT_CODEC_QOS_2M, _sdu, _rtn, _latency, _pd)
+
+/** @def BT_CODEC_QOS_OUT_FRAMED
+ *  @brief Helper to declare Output Framed bt_codec_qos
+ *
+ *  @param _interval SDU interval (usec)
+ *  @param _sdu Maximum SDU Size
+ *  @param _rtn Retransmission number
+ *  @param _latency Maximum Transport Latency (msec)
+ *  @param _pd Presentation Delay (usec)
+ */
+#define BT_CODEC_QOS_OUT_FRAMED(_interval, _sdu, _rtn, _latency, _pd) \
+	BT_CODEC_QOS(BT_CODEC_QOS_OUT, _interval, BT_CODEC_QOS_FRAMED, \
+		     BT_CODEC_QOS_2M, _sdu, _rtn, _latency, _pd)
+
+/** @def BT_CODEC_QOS_INOUT_FRAMED
+ *  @brief Helper to declare Output Framed bt_codec_qos
+ *
+ *  @param _interval SDU interval (usec)
+ *  @param _sdu Maximum SDU Size
+ *  @param _rtn Retransmission number
+ *  @param _latency Maximum Transport Latency (msec)
+ *  @param _pd Presentation Delay (usec)
+ */
+#define BT_CODEC_QOS_INOUT_FRAMED(_interval, _sdu, _rtn, _latency, _pd) \
+	BT_CODEC_QOS(BT_CODEC_QOS_INOUT, _interval, BT_CODEC_QOS_FRAMED, \
+		     BT_CODEC_QOS_2M, _sdu, _rtn, _latency, _pd)
+
+/** @brief Codec QoS structure. */
+struct bt_codec_qos {
+	/** QoS direction
+	 *
+	 *  This shall be set to BT_CODEC_QOS_OUT for broadcast sources, and
+	 *  shall be set to BT_CODEC_QOS_IN for broadcast sinks.
+	 */
+	uint8_t  dir;
```

Comment:
Doesn't necessarily have to be done in this PR, but unless the ordering of these struct members is critical, you could reduce the size of this struct by 32 bits by moving the `dir` member to the same 32-bit block that's occupied by `rtn` and `latency` (then there wouldn't be any unnecessary padding).